src/validation/mo_sethet/sethet.cpp

```diff
@@ -0,0 +1,202 @@
+// mam4xx: Copyright (c) 2022,
+// Battelle Memorial Institute and
+// National Technology & Engineering Solutions of Sandia, LLC (NTESS)
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <mam4xx/mam4.hpp>
+
+#include <mam4xx/aero_config.hpp>
+#include <skywalker.hpp>
+#include <validation.hpp>
+
+using namespace skywalker;
+using namespace mam4;
+using namespace haero;
+using namespace mo_sethet;
+
+void sethet(Ensemble *ensemble) {
+  ensemble->process([=](const Input &input, Output &output) {
+    using View1DHost = typename HostType::view_1d<Real>;
+    using ColumnView = haero::ColumnView;
+    constexpr int pver = mam4::nlev;
+    constexpr int gas_pcnst = mam4::gas_chemistry::gas_pcnst;
+
+    // non-ColumnView input values
+    const Real rlat = input.get_array("rlat")[0]; //need
+    //const Real rlat = -.2320924702;
+    const Real phis = input.get_array("phis")[0];
+    const Real delt = input.get_array("delt")[0];
+
+    const int spc_h2o2_ndx = input.get_array("spc_h2o2_ndx")[0] - 1;
+    const int spc_so2_ndx = input.get_array("spc_so2_ndx")[0] - 1;
+    const int h2o2_ndx = input.get_array("h2o2_ndx")[0] - 1;
+    const int so2_ndx = input.get_array("so2_ndx")[0] - 1;
+    const int h2so4_ndx = input.get_array("h2so4_ndx")[0] - 1;
+    const int gas_wetdep_cnt = input.get_array("gas_wetdep_cnt")[0];
+
+    int wetdep_map[3];
+    const auto wetdep_map_in = input.get_array("wetdep_map");
+    wetdep_map[0] = wetdep_map_in[0] - 1;
+    wetdep_map[1] = wetdep_map_in[1] - 1;
+    wetdep_map[2] = wetdep_map_in[2] - 1;
```

Comment:
I've got some indexing questions for you, @jeff-cohere @odiazib @mjs271, so the wetdep_map I'm testing with is from the [fortran validation data](https://github.com/eagles-project/mam_x_validation/blob/main/mo_sethet/mam_sethet_ts_355.py#L31). Are these indexes that should be -1ed before passing to the C++ code? What about the `*_ndx` indexes?

Reply:
I'm not familiar with these particular indices, but we've definitely been using 1-based indices in Fortran and 0-based indices in C++. Any indices that go into arrays must follow this convention--otherwise we're doing something Special and should explain why we're not following the usual language conventions.